src/Parser/Druid/Restoration/Modules/Talents/Photosynthesis.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import StatisticBox from 'Main/StatisticBox';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+const PHOTOSYNTHESIS_REJUV_INCREASE = 0.3;
+
+class Photosynthesis extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  rejuvenationIncrease = 0;
+  lifebloomIncrease = 0;
+
+  lastRealBloomTimestamp = null;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.PHOTOSYNTHESIS_TALENT.id);
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId === SPELLS.LIFEBLOOM_HOT_HEAL.id) {
+      this.lastRealBloomTimestamp = event.timestamp;
+    }
+  }
+
+  on_byPlayer_removebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId !== SPELLS.LIFEBLOOM_HOT_HEAL.id) {
+      return;
+    }
+    this.lastRealBloomTimestamp = event.timestamp;
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+    const amount = event.amount + (event.absorbed || 0);
+
+    if(spellId === SPELLS.REJUVENATION.id && this.combatants.selected.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id)) {
+      this.rejuvenationIncrease += amount * (1 - (1 / (1 + PHOTOSYNTHESIS_REJUV_INCREASE)));
+    }
+
+    if(spellId === SPELLS.LIFEBLOOM_BLOOM_HEAL.id && (this.lastRealBloomTimestamp === null || (event.timestamp - this.lastRealBloomTimestamp) > 32)){
```

Comment:
What's the magic number 32? I don't understand the logic of `lastRealBloomTimestamp`. I assume it's mean to track the last bloom heal that was not from Photosynthesis, but why is it set on debuff refreshed/removed? iirc the bloom happens every 15 sec after initial application (so if you refresh it it doesn't affect the bloom timing) reduced by haste, but it's been a while since I read up on it so could've changed. This makes it pretty hard to track due to dynamic haste scaling.

Reply:
Lifebloom also heal in the pandemic window. This has been taken from the legendary belt module, I assume it has been discussed why 32 is a fair window for accurate measuraments.

Reply:
hmm, might be better if @kfinch takes a look at it then, idk how that works. Unless lifebloom refreshes itself somehow when the belt procs (which seems weird) it doesn't make sense to me. Anyway should throw the 32 into a constant and call it MS_BUFFER or something so we know what it is  :P

Reply:
If you cast Lifebloom and your lifebloom has 12 seconds remaining, it triggers the bloom effect and refreshes the debuff.

I will throw the 32 ms into a constant

Reply:
The 32 ms was basically arbitrary, I think Zero came up with it months ago when he was looking for the gap between some two events that should happen together, and the biggest gap he saw was 32 ms. In practice it varies by spell/event pair. Some are always on the same timestamp, while for other effects I've observed gaps of as much as 150ms.

When picking a buffer I've personally often defaulted to 50, but will make it larger if it's something where there is almost no chance of a false positive.

Anyway, 32 should be fine for this.